packages/popover/src/vaadin-popover.js

```diff
@@ -532,7 +532,13 @@ class Popover extends PopoverPositionMixin(
 
     document.documentElement.removeEventListener('click', this.__onGlobalClick, true);
 
-    this._openedStateController.close(true);
+    // Automatically close popover when it is removed from DOM
+    // Avoid closing if the popover is just moved in the DOM
+    queueMicrotask(() => {
+      if (!this.isConnected) {
+        this._openedStateController.close(true);
+      }
+    });
```

Comment:
This is adopted from notification: https://github.com/vaadin/web-components/blob/003bbb39e65e3b7a60d609dadb8bc1d9d418b33b/packages/notification/src/vaadin-notification-mixin.js#L255-L259